### ISSUE TITLE
Unexpected Behavior: Server Side Function running twice

### ISSUE THREAD

**Issue by Harshkt214** [vercel/next.js#69576]
### Link to the code that reproduces this issue

https://github.com/Harshkt214/next-bug

### To Reproduce

1. Clone repo and build application using `pnpm build`.
2. Start production server locally `pnpm start`
3. Make get request to `http://localhost:3000/account/[user]`.
4. See the server console output.

### Current vs. Expected behavior

```tsx
export const dynamic = "force-static";

const getData = async () => {
   console.log("getData ", Date.now());
   return null;
};
export default async function User({ params }: { params: { user: string } }) {
   const data = await getData();
   console.log(data);
   console.log(Date.now());
   return <>{JSON.stringify(params, null, 4)}</>;
}
```
I used above code to build static page on-demand when user visit the page instead of build time (to decrease next js production build time) but when I make request in production mode (http://localhost:3000/account/user) the function was running twice producing the output

![image](https://github.com/user-attachments/assets/78105378-b7ec-4187-9094-266ffb844582)

I was expecting it to be run only once, however it was running once in SSR when removed `export const dynamic = "force-static"`.

### Provide environment information

```bash
Operating System:
	Platform: windows
	Version: windows 10
Binaries:
	Node: 22.7.0
	npm: 10.8.2
	pnpm: 9.2.0
Relevant Packages:
	next : 14.2.7
	eslint-config-next:14.2.7
	react : ^18
	react-dom: ^18
	typescript: ^5
```

### Which area(s) are affected? (Select all that apply)

Output (export/standalone)

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed)

**Comment by gianlucadifrancesco** [comment]
Hi @Harshkt214!
I think this has already been addressed and fixed in #67680, released in [v15.0.0-canary.70](https://github.com/vercel/next.js/releases/tag/v15.0.0-canary.70).

I tried to upgrade your project with `pnpm i next@15.0.0-canary.70 react@rc react-dom@rc` and now it works, could you please confirm it's solved for you as well?

**Comment by Harshkt214** [comment]
Hello @gianlucadifrancesco, thanks for your reply.
I confirm it is working in `next@15.0.0-canary.70` and above versions as expected.
